Modules/IndividualAssessment/classes/ilIndividualAssessmentUserGrading.php

```diff
@@ -226,20 +237,29 @@ public function toFormInput(
 
                 $updated_custom = [];
                 foreach ($custom_fields as $cf) {
-                    $updated_custom[] = $cf->withValue($values['custom'][$cf->getFieldId()]);
+                    $value = $values['custom'][$cf->getFieldId()];
+                    if ($cf->hasNotes()) {
+                        list($value, $note) = $value;
+                        $cf = $cf->withNote($note);
+                    }
+                    $updated_custom[] = $cf->withValue($value);
+                }
+
+                $learning_progress = $this->getLearningProgress();
+                if ($manual_grading) {
+                    $learning_progress = (int) $values['learning_progress'];
                 }
 
                 return (new ilIndividualAssessmentUserGrading(
                     $values['name'],
-                    $values['record'],
-                    $values['internal_note'],
-                    $file,
-                    (int) $values['learning_progress'],
-                    $values['place'],
-                    $values['event_time'],
+                    $values['record'] ?? '',
+                    $values['internal_note'] ?? '',
+                    $file ?? null,
+                    $learning_progress,
+                    $values['place'] ?? '',
+                    $values['event_time'] ?? null,
```

Comment:
Ich mag diese transforms nicht; irgendwie "lügt" die Klasse damit. Wenn das alles leer sein darf, würde ich eher die Parameter in ilIndividualAssessmentUserGrading auf optional setzen, als mit pseudo-values irgendwie den construct zu erfüllen.